nomad/structs/volumes.go

```diff
@@ -100,12 +100,11 @@ func (v *VolumeRequest) Copy() *VolumeRequest {
 	nv := new(VolumeRequest)
 	*nv = *v
 
-	if v.MountOptions == nil {
-		return nv
+	if v.MountOptions != nil {
+		mo := *v.MountOptions
+		nv.MountOptions = &mo
 	}
```

Comment:
Looking again, this MonutOptions should be copied deeply not just the reference:

```suggestion
	nv.MountOptions = v.MountOptions.Copy()
```

Will fix: